modules/schedule/index.js

```diff
@@ -0,0 +1,345 @@
+        if(err) console.log("Error writing command file: " + err);
+  });
+}
+
+function newSchedule(data) {
+  var s;
+
+  try {
+    s = later.parse.text(data['schedule']);
+  } catch (ex) {
+    return ex;
+  }
+
+  if(typeof(s) == "number" || s.error >= 0 )
+  {
+    var m = "Provided schedule query doesn't parse:\n";
+    var offset = 0;
+
+    if( typeof(s) == "number")
+      offset = s;
+    else
+      offset = s.error;
+    
+    m += "    " + data['schedule'].slice(0,offset) + '\u032D' 
+         + data['schedule'].slice(offset,data['schedule'].length);
+    return m;
+  }
+  
+  data['schedule'] = s;
+
+  // check that command is valid
+  //   - check if commands are disabled
+  //   - doesn't call schedule
+  //   - doesn't call a dumbcommand which calls schedule
+  //     * checked via perceived caller
+  //   - isn't too close to last call
+  // check execution frequency is below some minimum (config)
+  
+  if(noCommands && data.command.substring(0, bot.config.commandPrefix.length) 
+      == bot.config.commandPrefix) 
+    return "Commands are disabled";
+
+  if(data['command'].match(/^!schedule/))
+    return data['blame'] + ": fuck you"; 
+  
+  if(data['blame'] == bot.client.nick)
+    return "Cannot call schedule recursively.";
+  
+  if(minimumCreationDelay > 0 && schedules.length > 0 
+      && moment().diff(schedules.slice(-1)[0].created, 'seconds') <= minimumCreationDelay)
+    return "Schedule creation is rate limited. Please wait at least " + minimumCreationDelay
+           + " seconds between schedule creation.";
+
+  //s.range should be the range, in seconds, between schedule calls...
+  //but it's undefined.
+  var interval = getAverageInterval(s);
+  if(minimumInterval > 0 && interval < minimumInterval)
+    return "Parsed average frequency of " + moment.duration(interval, "seconds").format() + " is below the minimum "
+           + "interval of " + minimumInterval + " seconds";
+
+  // Handle private message schedules. These show as channel being us--- it should
+  // be them.
+  if(data.channel == bot.client.nick)
+    data.channel = data.blame;
+
+  // FIXME: This is awful. Asynchronous shit makes this a baaad idea.
+  schedules.push(data);
+  registerCommand(data);
+
+  writeSchedule(schedules);
+  
+  var next = getDifference(later.schedule(s).next(1), moment()).format();
+  if(next.match(/^\d+$/))
+    next = next + " seconds"; //FIXME: vv breaks
+
+  if(next == "0") //FIXME: this usually isn't true. It may say now, but it never makes it.
+    return "Created, first execution is now.";
+  else
+    return "Created, the first execution is in " + next + ".";
+}
+
+function registerCommand(data) {
+  var command;
+
+  if(data.calls == 0)
+  {
+    //see FIXME below
+    timers.push(undefined);
+    return;
+  }
+
+  command = function() {
+     // deactivate command
+     if(data.calls == 0)
+     {
+       var i = schedules.indexOf(data);
+
+       //FIXME: this is awful, and will lead to a large array
```

Comment:
Couldn't you remove it from both at the same time here?

Alternately, maybe move it over into a "deadSchedules" thing if you want it to still be visible for some duration after expiring? I might not be understanding the point of this.